yazi-core/src/input/snaps.rs

```diff
@@ -11,11 +11,15 @@ pub(super) struct InputSnaps {
 
 impl InputSnaps {
 	#[inline]
-	pub(super) fn reset(&mut self, value: String, limit: usize) {
+	pub(super) fn reset(&mut self, value: String, limit: usize, cursor_at: Option<usize>) {
 		self.idx = 0;
 		self.versions.clear();
 		self.versions.push(InputSnap::new(value, limit));
 		self.current = self.versions[0].clone();
+
+		if let Some(cursor) = cursor_at {
```

Comment:
Why do we need this?

Reply:
So that if no cursor position is specified it will be placed at the end